src/lib.rs

```diff
@@ -62,10 +62,46 @@ mod utils;
 
 pub use datatypes::{RepeatMode, SpeakerInfo};
 pub use discovery::{discover, find};
+pub use rupnp::{self, ssdp::URN};
 pub use snapshot::Snapshot;
 pub use speaker::Speaker;
+use thiserror::*;
 pub use track::{Track, TrackInfo};
 
-pub use rupnp::{self, ssdp::URN, Error};
+/// Represents an error encountered by Sonor
+#[derive(Error, Debug)]
+pub enum Error {
+    /// Errors sourced from the rupnp crate
+    #[error(transparent)]
+    UPnP(#[from] rupnp::Error),
+    /// Errors sourced from XML parsing
+    #[error(transparent)]
+    Xml(#[from] roxmltree::Error),
```

Comment:
Would it make sense to reuse the `Error::UPnP(rupnp::Error::XmlError(..))` instead of an own `Error::Xml` variant? I can't think of a case where you'd want to handle those separately.

Reply:
I arrived at this list by seeing which type conversions the compiler claimed were missing; my recollection for this one is that `sonor` is doing something more directly with xml parsing and that this specific type conversion was required; it wasn't actually a UPnP error because it wasn't generated by the upnp crate.

Reply:
I see.